lib/services/docker_container_initializer.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2020 - present Kurtosis Technologies LLC.
+ * All Rights Reserved.
+ */
+
+package services
+
+import "os"
+
+// GENERIC TOOD: If Go had generics, this would be parameterized with the subtype of Service that this returns
+type DockerContainerInitializer interface {
```

Comment:
With the exception of `GetDockerImage`, this is all copied from the old `ServiceInitializerCore`

Reply:
i'm a huuuge fan of this rename